build-info-extractor-gradle/src/main/groovy/org/jfrog/gradle/plugin/artifactory/extractor/listener/ProjectsEvaluatedBuildListener.groovy

```diff
@@ -131,8 +132,16 @@ public class ProjectsEvaluatedBuildListener extends BuildAdapter implements Proj
      * @param publicationsNames - Publications names separated by commas
      */
     private static void addPublications(ArtifactoryTask artifactoryTask, PublishingExtension publishingExtension, String publicationsNames) {
+        PublicationContainer container = publishingExtension.getPublications();
         for (publicationName in publicationsNames.split(",")) {
-            Publication publication = publishingExtension.getPublications().findByName(publicationName);
+            // If ALL_PUBLICATIONS parameter was provided, add all publications and return.
+            if (publicationName == TaskHelperPublications.ALL_PUBLICATIONS) {
```

Comment:
I'm not sure if the following may introduce an issue or not, but even if doesn't, it may cause issues in the future, following changes to the code, so I syggest changing it to be on the safe side.
This code may cause the same publication to be added multiple times to the Artifactory Plugin's publications collection.
I suggest storing all the publications collected in the case of ALL_PUBLICATIONS in a different collction, then the collection used for storing the publications which names were specified specifically. This should allow saving either the ALL_PUBLICATIONS publications, or the specific ones.

Reply:
I improved the code there to avoid redundant additions of publications when ALL_PUBLICATIONS is passed.
The publications are kept in a set so will be unique either way.
I think adding another collection will complicate the code a bit, since currently the CI and plugin share the same collections which is convenient.